docs/developer-docs/security/rust-canister-development-security-best-practices.md

```diff
@@ -461,6 +461,99 @@ Using the rust CDK, the reccuring timer is also lost on upgrade as explained in
+
+#### Recommendation
+
+Make sure the endpoints, called by an HTTP outcall, are idempotent, i.e. the queried endpoint has the same behaviour with the same request payload, no matter the number of times it is called.
+
+Some servers support the use of idempotency keys. These keys are random unique strings submitted in the HTTP request as headers. If used with the HTTP outcalls feature, all requests sent by each (honest) replica will contain the same idempotency key. This allows the server to recognize duplicated requests (i.e requests with the same idempotency key), handle just one and modify the server state only once. Note that this is a feature that must be supported by the server.
+
+[More information.](https://internetcomputer.org/docs/current/developer-docs/integrations/https-outcalls/https-outcalls-how-it-works#post-requests-must-be-idempotent)
+
+### Ensure HTTP responses are identical
+
+#### Security concern
+
+When replicas of a subnet receive HTTP responses, these responses must be identical. Otherwise the consensus won’t be achieved and the HTTP response will be rejected, but still charged.
+
+#### Recommendation
+
+Make sure the HTTP responses sent to the consensus layer are identical.
+
+Ideally the HTTP responses returned by the queried endpoint would be always the same. However most of the time this is not possible to control and the responses include random data (e.g the response includes timestamps, cookie values or some sort of identifiers). In those cases make sure to use the [transformation functions](https://internetcomputer.org/docs/current/developer-docs/integrations/https-outcalls/https-outcalls-how-it-works#transformation-function) to guarantee that the responses, received by each replica, are identical by removing any random data or extracting only the relevant data.
+
+This applies to the HTTP response body and header. Make sure to consider both when applying the transformation functions. Response headers are often overlooked and lead to failure because of failed consensus.
+
+[More information.](https://internetcomputer.org/docs/current/developer-docs/integrations/https-outcalls/https-outcalls-how-it-works#responses-must-be-similar)
+
+### Be aware of HTTP request and response sizes
+
+#### Security concern
+
+HTTP outcalls [pricing](https://internetcomputer.org/docs/current/developer-docs/integrations/https-outcalls/https-outcalls-how-it-works#pricing) is determined by, among other variables, the size of the HTTP request and the maximal response size. Thus, if big requests are made, this could quickly drain the canister’s cycles balance. This can be risky e.g. HTTP outcalls are triggered by user actions (rather than a heartbeat or timer invocation).
```

Comment:
nit: 
```suggestion
HTTP outcalls [pricing](https://internetcomputer.org/docs/current/developer-docs/integrations/https-outcalls/https-outcalls-how-it-works#pricing) is determined by, among other variables, the size of the HTTP request and the maximal response size. Thus, if big requests are made, this could quickly drain the canister’s cycles balance. This can be risky e.g. if HTTP outcalls are triggered by user actions (rather than a heartbeat or timer invocation).
```